### ISSUE TITLE
avg calculation with Parquet file in vector runtime seems more "off"

### ISSUE THREAD

**Issue by philrz** [brimdata/super#5530]
## tl;dr

This is a follow-on from #5516. Repeating the `avg` calculation using the test data from that issue, consider the results in the table below. Treating the result `1058.5234017218875` from DuckDB as a baseline, with the different super formats and runtime options we see:

|**Format**|**Runtime**|**Result**|**Delta**|
|-|-|-|-|
|BSUP|sequential|`1058.523401720017`|`0.0000000018705`|
|CSUP|sequential|`1058.523401720017`|`0.0000000018705`|
|Parquet|sequential|`1058.523401720017`|`0.0000000018705`|
|CSUP|vector|`1058.5234017218877`|`0.0000000000002`|
|Parquet|vector|`1058.5297770606794`|`0.0063753387919`|

## Details

Repro is with super commit 883ffd2.

The the runs that generated the results in the table above are in https://github.com/brimdata/super/issues/5516#issuecomment-2539695424 and the test data is linked from that issue.

Users are accustomed to seeing small differences in precision with floating point math, so I expect this might all be explained by something about parallel operations. Indeed, we've seen some non-deterministic floating point calculations with other database systems, for instance. However, seeing the delta as soon as the 4th digit with vector Parquet is surprising compared to out at 9+ digits like the others, so I figured I'd surface this in case it's worthy of closer scrutiny.

**Comment by philrz** [comment]
I've verified in super commit 6300fba that the partial fix in linked PR #5681 does indeed tighten up the delta. Repeating the query against Parquet, we now see:

```
$ super -version
Version: v1.18.0-304-g6300fbaf

$ SUPER_VAM=1 super -c 'from data.parquet | avg(event_value)'
1058.5250095975998
```

So that delta is `0.00160788` which is ~4x less than the prior `0.0063753387919`.

**Comment by philrz** [comment]
Verified in super commit cc7925e.

Repeating the query against Parquet, we now see:

```
$ super -version
Version: v1.18.0-459-gcc7925ea6

$ SUPER_VAM=1 super -c 'from data.parquet | avg(event_value)'
1058.5234017218872
```

So the delta is now only `0.0000000000003`, which is very much within the tiny range of what's typically seen for deltas between floating point calculations in different systems.

Thanks @nwt!
